relay/blockchain/currency_network_proxy.py

```diff
@@ -48,8 +46,8 @@ class CurrencyNetworkProxy(Proxy):
                             TrustlineRequestEventType,
                             TrustlineUpdateEventType]
 
-    def __init__(self, web3, abi, address: str) -> None:
-        super().__init__(web3, abi, address)
+    def __init__(self, web3, abi, address: str, config) -> None:
```

Comment:
Dont pass the whole config object, as it introduces a dependency. Just pass the needed value as a default parameter.